src/pages/scientificServices/pipelines/components/inputs/PipelineFloatInput.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { ValidatedInput } from 'src/components/input';
+import { PipelineInput } from 'src/libs/ajax/teaspoons/teaspoons-models';
+import { INPUT_DESCRIPTIONS } from 'src/pages/scientificServices/pipelines/utils/pipeline-input-utils';
+
+interface PipelineFloatInputProps {
+  input: PipelineInput;
+  value: string;
+  onChange: (value: string) => void;
+  validationError?: string;
+  onValidation(error?: string): void;
+}
+
+export const PipelineFloatInput: React.FC<PipelineFloatInputProps> = ({
+  input,
+  value,
+  onChange,
+  validationError,
+  onValidation,
+}) => {
+  const { label, placeholder, helpText } = INPUT_DESCRIPTIONS[input.name];
+
+  return (
+    <>
+      <h3 style={{ marginBottom: '0.5rem' }}>
+        {label || input.name}{' '}
+        {input.isRequired ? (
+          <span style={{ color: '#DB3214' }}>*</span>
+        ) : (
+          <span style={{ fontStyle: 'italic', fontWeight: 'normal' }}> - optional</span>
+        )}
+      </h3>
+      <ValidatedInput
+        width={400}
+        error={validationError}
+        inputProps={{
+          'aria-label': `${input.name} text input`,
+          type: 'text',
```

Comment:
`ValidatedInput` is a wrapper component for the low-level HTML `input` element, which accepts input types like `text`, `month`, `number`, etc, so you can do something like this:

```suggestion
          'aria-label': `${input.name} number input`,
          type: 'number',
```

this will handle validating that the user doesn't enter in any ordinary string, so you wouldn't need the float regex validation below.

Reply:
thank you!